dagql/idtui/frontend_plain.go

```diff
@@ -259,17 +278,15 @@ func (fe *frontendPlain) render() {
 }
 
 func (fe *frontendPlain) finalRender() {
-	if fe.Debug || fe.Verbosity > 0 {
-		fe.render()
-	}
```

Comment:
@jedevc - pretty sure this conditional didn't quite make sense for pretty output. I'm assuming it took this logic from the TUI, but the TUI only had it because in that case it was printing the _full_ progress output. In this case the final `render()` call is needed in order to not miss the last bit of progress. Without this fix, the `too bad` test was flaky/failing.

Reply:
Hm, you're right, not quite sure what I was thinking.

However, (just noticed this) we should ideally track the equivalent of `fe.err` - otherwise, we don't show any progress output on an error with `--progress=plain --silent`. Hopefully that makes sense.

We'd also want to set `contextHold = 0` here, if that's the case :thinking:

Reply:
I just pushed a fix for `--silent` - removing these conditionals actually made it print progress on exit with `--silent` which seems like very much not what you want. :smile: 

Not sure about `contextHold = 0` and `fe.err`, but I tried running things that errored with `--silent` and it seems OK. :thinking: Lemme know if this is blocking, or feel free to push changes to this PR :pray: I think you have access

Reply:
Aha, we don't need `fe.err`, but I pushed a change to `contextHold` - that's just to disable the default delay in how we do "switching" between different contexts.